### PR TITLE
[AIR/Train] `HuggingFacePredictor` improvements

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

Adds `to_air_checkpoint` method to Hugging Face, improves `HuggingFacePredictor` by clarifying APIs & docstrings.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/train/huggingface/utils.py [hunk 1]
```diff
@@ -19,6 +21,42 @@
...
+    preprocessor: Optional["Preprocessor"] = None,
+    path: Optional[str] = None,
+) -> Checkpoint:
+    """Convert a pretrained Transformers model to AIR checkpoint for serve or inference.
+
+    Args:
+        model: Either a ``transformers.PreTrainedModel``, or a trained PyTorch model.
+        tokenizer: Tokenizer to use.
+        preprocessor: A fitted preprocessor. The preprocessing logic will
+            be applied to serve/inference.
```
Comment (amogkam) [inline python/ray/train/huggingface/utils.py:37]:
```suggestion
            be applied to the inputs for serving/inference.
```

File: python/ray/train/huggingface/utils.py [hunk 2]
```diff
@@ -19,6 +21,42 @@
...
+    model: Union[transformers.modeling_utils.PreTrainedModel, torch.nn.Module],
+    tokenizer: Optional[transformers.PreTrainedTokenizer] = None,
+    preprocessor: Optional["Preprocessor"] = None,
+    path: Optional[str] = None,
+) -> Checkpoint:
+    """Convert a pretrained Transformers model to AIR checkpoint for serve or inference.
+
+    Args:
+        model: Either a ``transformers.PreTrainedModel``, or a trained PyTorch model.
+        tokenizer: Tokenizer to use.
```
Comment (amogkam) [inline python/ray/train/huggingface/utils.py:35]: Is there any more information we can provide on the tokenizer/how it will be used?

**Review by amogkam — APPROVED** [review]
Nice, thanks @Yard1!

File: python/ray/train/huggingface/utils.py [hunk 3]
```diff
@@ -19,6 +21,43 @@
...
+) -> Checkpoint:
+    """Convert a pretrained Transformers model to AIR checkpoint for serve or inference.
+
+    Args:
+        model: Either a ``transformers.PreTrainedModel``, or a trained PyTorch model.
+        tokenizer: Tokenizer to be used in the Transformers pipeline
+            during serving/inference.
+        preprocessor: A fitted preprocessor. The preprocessing logic will
+            be applied to the inputs for serving/inference.
+        path: The directory where the checkpoint will be stored to.
```
Comment (xwjiang2010) [inline python/ray/train/huggingface/utils.py:39]: whose responsibility it is to delete this path in the future if it's temporary?

Comment (xwjiang2010) [reply python/ray/train/huggingface/utils.py:39]: is it usually the case that `model` is coming from a file already and we are writing it back to another file?

Comment (xwjiang2010) [reply python/ray/train/huggingface/utils.py:39]: I would imagine if the user doesn't supply anything, they wouldn't want to manage it either. Then it's left to nobody to clean it up and may blow up the disk.

Comment (Yard1) [reply python/ray/train/huggingface/utils.py:39]: I'll use explicit path for now, let's discuss later!

**Review by xwjiang2010 — APPROVED** [review]
Thanks so much!!

**Pull request merged**